Not working when a directory has a space

This package does not work if there is a space in your path (OS: Windows)

Repro:
1. Use Windows
2. Clone Stryker in a folder with a path
3. `npm install`
4. `npm test`

Output:
```
  1) we have a module using stryker                                                                     
       after installing Stryker                                                                         
         "before all" hook:                                                                             
     Command failed: npm install                                                                        
{ Error: Command failed: npm pack C:\dev\github repos\stryker\packages\stryker-api                      
npm ERR! code ENOLOCAL                                                                                  
npm ERR! Could not install from "..\..\..\..\..\dev\github" as it does not contain a package.json file. 
                                                                                                        
npm ERR! A complete log of this run can be found in:                                                    
npm ERR!     C:\Users\simonl\AppData\Roaming\npm-cache\_logs\2018-11-19T15_44_58_657Z-debug.log         
                                                                                                        
    at ChildProcess.exithandler (child_process.js:291:12)                                               
    at ChildProcess.emit (events.js:182:13)                                                             
    at maybeClose (internal/child_process.js:961:16)                                                    
    at Process.ChildProcess._handle.onexit (internal/child_process.js:248:5)                            
  killed: false,                                                                                        
  code: 1,                                                                                              
  signal: null,                                                                                         
  cmd:                                                                                                  
   'npm pack C:\\dev\\github repos\\stryker\\packages\\stryker-api' }                                   
npm ERR! code ELIFECYCLE                                                                                
npm ERR! errno 1                                                                                        
npm ERR! test-module@0.0.0 postinstall: `install-local`                                                 
npm ERR! Exit status 1                                                                                  
npm ERR!                                                                                                
npm ERR! Failed at the test-module@0.0.0 postinstall script.                                            
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.      
                                                                                                        
npm ERR! A complete log of this run can be found in:                                                    
npm ERR!     C:\Users\simonl\AppData\Roaming\npm-cache\_logs\2018-11-19T15_44_58_722Z-debug.log         
                                                                                                        
  Error: Command failed: npm install                                                                    
  { Error: Command failed: npm pack C:\dev\github repos\stryker\packages\stryker-api                    
  npm ERR! code ENOLOCAL                                                                                
  npm ERR! Could not install from "..\..\..\..\..\dev\github" as it does not contain a package.json file
.                                                                                                       
                                                                                                        
  npm ERR! A complete log of this run can be found in:                                                  
  npm ERR!     C:\Users\simonl\AppData\Roaming\npm-cache\_logs\2018-11-19T15_44_58_657Z-debug.log       
                                                                                                        
      at ChildProcess.exithandler (child_process.js:291:12)                                             
      at maybeClose (internal/child_process.js:961:16)                                                  
      at Process.ChildProcess._handle.onexit (internal/child_process.js:248:5)                          
    killed: false,                                                                                      
    code: 1,                                                                                            
    signal: null,                                                                                       
    cmd:                                                                                                
     'npm pack C:\\dev\\github repos\\stryker\\packages\\stryker-api' }                                 
  npm ERR! code ELIFECYCLE                                                                              
  npm ERR! errno 1                                                                                      
  npm ERR! test-module@0.0.0 postinstall: `install-local`                                               
  npm ERR! Exit status 1                                                                                
  npm ERR!                                                                                              
  npm ERR! Failed at the test-module@0.0.0 postinstall script.                                          
  npm ERR! This is probably not a problem with npm. There is likely additional logging output above.    
                                                                                                        
  npm ERR! A complete log of this run can be found in:                                                  
  npm ERR!     C:\Users\simonl\AppData\Roaming\npm-cache\_logs\2018-11-19T15_44_58_722Z-debug.log       
                                                                                                        
      at ChildProcess.exithandler (child_process.js:291:12)                                             
      at maybeClose (internal/child_process.js:961:16)                                                  
      at Process.ChildProcess._handle.onexit (internal/child_process.js:248:5)                          
                                                                                                        
                                                                                                        
                                                                                                        
                                                                                                        
    at Promise.all.then.arr (C:\dev\github repos\stryker\node_modules\lerna\node_modules\execa\index.js:
236:11)                                                                                                 
    at process._tickCallback (internal/process/next_tick.js:68:7)                                       
  code: 1,                                                                                              
  killed: false,                                                                                        
  stdout:                                                                                               
   '\n> stryker-api@0.21.5 test C:\\dev\\github repos\\stryker\\packages\\stryker-api\n> nyc --reporter=
html --report-dir=reports/coverage --check-coverage --lines 90 --functions 68 --branches 63 npm run moch
a\n\n\n> stryker-api@0.21.5 mocha C:\\dev\\github repos\\stryker\\packages\\stryker-api\n> mocha "test/u
nit/**/*.js" && mocha --timeout 100000 "test/integration/**/*.js"\n\n\n\n  Config\n    defaults\n      √
 should set default thresholds as expected\n    set\n      √ should override values\n      √ should over
ride thresholds when assigned\n      √ should never clear thresholds\n\n  Factory\n    when creating a s
ub-class\n      √ should have empty known names\n      √ should throw an error if it is requested to cre
ate a non-existing item\n      when registering a test class "some-item"\n        √ should retrieve a ne
w item when `create` is called with "some-item"\n\n  File\n    √ should allow utf8 encoded string conten
t in the constructor\n    √ should allow buffered content in the constructor\n    √ should be serializab
le\n\n  LogLevel\n    √ should provide "off" for log level "off"\n    √ should provide "fatal" for log l
evel "fatal"\n    √ should provide "error" for log level "error"\n    √ should provide "warn" for log le
vel "warn"\n    √ should provide "info" for log level "info"\n    √ should provide "debug" for log level
 "debug"\n    √ should provide "trace" for log level "trace"\n\n  MutantFactory\n    when correct mutato
r is not installed\n      √ should throw an error with package name "stryker-javascript-mutator" when us
ing the "javascript" mutator\n      √ should throw an error with package name "stryker-vue-mutator" when
 using the "vue" mutator\n      √ should throw an error with package name "stryker-typescript" when usin
g the "typescript" mutator\n\n\n  20 passing (26ms)\n\n\n\n  we have a module using stryker\n    after i
nstalling Stryker\nExec \'npm install\' in C:\\dev\\github repos\\stryker\\packages\\stryker-api\\testRe
sources\\module\n\n> test-module@0.0.0 postinstall C:\\dev\\github repos\\stryker\\packages\\stryker-api
\\testResources\\module\n> install-local\n\n\r\n[install-local] packing - 0/1\n      1) "before all" hoo
k\n\n\n  0 passing (7s)\n  1 failing\n\n  1) we have a module using stryker\n       after installing Str
yker\n         "before all" hook:\n     Command failed: npm install\n{ Error: Command failed: npm pack C
:\\dev\\github repos\\stryker\\packages\\stryker-api\nnpm ERR! code ENOLOCAL\nnpm ERR! Could not install
 from "..\\..\\..\\..\\..\\dev\\github" as it does not contain a package.json file.\n\nnpm ERR! A comple
te log of this run can be found in:\nnpm ERR!     C:\\Users\\simonl\\AppData\\Roaming\\npm-cache\\_logs\
\2018-11-19T15_44_58_657Z-debug.log\n\n    at ChildProcess.exithandler (child_process.js:291:12)\n    at
 ChildProcess.emit (events.js:182:13)\n    at maybeClose (internal/child_process.js:961:16)\n    at Proc
ess.ChildProcess._handle.onexit (internal/child_process.js:248:5)\n  killed: false,\n  code: 1,\n  signa
l: null,\n  cmd:\n   \'npm pack C:\\\\dev\\\\github repos\\\\stryker\\\\packages\\\\stryker-api\' }\nnpm
 ERR! code ELIFECYCLE\nnpm ERR! errno 1\nnpm ERR! test-module@0.0.0 postinstall: `install-local`\nnpm ER
R! Exit status 1\nnpm ERR! \nnpm ERR! Failed at the test-module@0.0.0 postinstall script.\nnpm ERR! This
 is probably not a problem with npm. There is likely additional logging output above.\n\nnpm ERR! A comp
lete log of this run can be found in:\nnpm ERR!     C:\\Users\\simonl\\AppData\\Roaming\\npm-cache\\_log
s\\2018-11-19T15_44_58_722Z-debug.log\n\n  Error: Command failed: npm install\n  { Error: Command failed
: npm pack C:\\dev\\github repos\\stryker\\packages\\stryker-api\n  npm ERR! code ENOLOCAL\n  npm ERR! C
ould not install from "..\\..\\..\\..\\..\\dev\\github" as it does not contain a package.json file.\n  \
n  npm ERR! A complete log of this run can be found in:\n  npm ERR!     C:\\Users\\simonl\\AppData\\Roam
ing\\npm-cache\\_logs\\2018-11-19T15_44_58_657Z-debug.log\n  \n      at ChildProcess.exithandler (child_
process.js:291:12)\n      at maybeClose (internal/child_process.js:961:16)\n      at Process.ChildProces
s._handle.onexit (internal/child_process.js:248:5)\n    killed: false,\n    code: 1,\n    signal: null,\
n    cmd:\n     \'npm pack C:\\\\dev\\\\github repos\\\\stryker\\\\packages\\\\stryker-api\' }\n  npm ER
R! code ELIFECYCLE\n  npm ERR! errno 1\n  npm ERR! test-module@0.0.0 postinstall: `install-local`\n  npm
 ERR! Exit status 1\n  npm ERR! \n  npm ERR! Failed at the test-module@0.0.0 postinstall script.\n  npm 
ERR! This is probably not a problem with npm. There is likely additional logging output above.\n  \n  np
m ERR! A complete log of this run can be found in:\n  npm ERR!     C:\\Users\\simonl\\AppData\\Roaming\\
npm-cache\\_logs\\2018-11-19T15_44_58_722Z-debug.log\n  \n      at ChildProcess.exithandler (child_proce
ss.js:291:12)\n      at maybeClose (internal/child_process.js:961:16)\n      at Process.ChildProcess._ha
ndle.onexit (internal/child_process.js:248:5)\n\n\n\n',                                                 
  stderr:                                                                                               
   '{ Error: Command failed: npm pack C:\\dev\\github repos\\stryker\\packages\\stryker-api\nnpm ERR! co
de ENOLOCAL\nnpm ERR! Could not install from "..\\..\\..\\..\\..\\dev\\github" as it does not contain a 
package.json file.\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     C:\\Users\\simo
nl\\AppData\\Roaming\\npm-cache\\_logs\\2018-11-19T15_44_58_657Z-debug.log\n\n    at ChildProcess.exitha
ndler (child_process.js:291:12)\n    at ChildProcess.emit (events.js:182:13)\n    at maybeClose (interna
l/child_process.js:961:16)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:248:5)
\n  killed: false,\n  code: 1,\n  signal: null,\n  cmd:\n   \'npm pack C:\\\\dev\\\\github repos\\\\stry
ker\\\\packages\\\\stryker-api\' }\nnpm ERR! code ELIFECYCLE\nnpm ERR! errno 1\nnpm ERR! test-module@0.0
.0 postinstall: `install-local`\nnpm ERR! Exit status 1\nnpm ERR! \nnpm ERR! Failed at the test-module@0
.0.0 postinstall script.\nnpm ERR! This is probably not a problem with npm. There is likely additional l
ogging output above.\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     C:\\Users\\si
monl\\AppData\\Roaming\\npm-cache\\_logs\\2018-11-19T15_44_58_722Z-debug.log\n\nnpm ERR! code ELIFECYCLE
\nnpm ERR! errno 1\nnpm ERR! stryker-api@0.21.5 mocha: `mocha "test/unit/**/*.js" && mocha --timeout 100
000 "test/integration/**/*.js"`\nnpm ERR! Exit status 1\nnpm ERR! \nnpm ERR! Failed at the stryker-api@0
.21.5 mocha script.\nnpm ERR! This is probably not a problem with npm. There is likely additional loggin
g output above.\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     C:\\Users\\simonl\
\AppData\\Roaming\\npm-cache\\_logs\\2018-11-19T15_44_58_801Z-debug.log\nnpm ERR! code ELIFECYCLE\nnpm E
RR! errno 1\nnpm ERR! stryker-api@0.21.5 test: `nyc --reporter=html --report-dir=reports/coverage --chec
k-coverage --lines 90 --functions 68 --branches 63 npm run mocha`\nnpm ERR! Exit status 1\nnpm ERR! \nnp
m ERR! Failed at the stryker-api@0.21.5 test script.\nnpm ERR! This is probably not a problem with npm. 
There is likely additional logging output above.\n\nnpm ERR! A complete log of this run can be found in:
\nnpm ERR!     C:\\Users\\simonl\\AppData\\Roaming\\npm-cache\\_logs\\2018-11-19T15_44_59_160Z-debug.log
\n',                                                                                                    
  failed: true,                                                                                         
  signal: null,                                                                                         
  cmd: 'npm run test',                                                                                  
  timedOut: false,                                                                                      
  exitCode: 1 }                                                                                         
npm ERR! Test failed.  See above for more details.                                                      
C:\dev\github repos\stryker [master ≡]                                                                  
```

Found by @Wmaarts

--- a/src/LocalInstaller.ts
+++ b/src/LocalInstaller.ts
@@ -24,6 +24,10 @@ export interface ListByPackage {
 }
 
 const TEN_MEGA_BYTE = 1024 * 1024 * 10;
+
+function quotify(value: string) {
+    return `"${value}"`;
+}
 export class LocalInstaller extends EventEmitter {
     private sourcesByTarget: ListByPackage;
     private options: Options;
@@ -81,7 +85,10 @@ export class LocalInstaller extends EventEmitter {
     }
 
     private installOne(target: InstallTarget): Promise<void> {
-        const toInstall = target.sources.map(source => resolvePackFile(this.uniqueDir, source.packageJson)).join(' ');
+        const toInstall = target.sources
+            .map(source => resolvePackFile(this.uniqueDir, source.packageJson))
+            .map(quotify)
+            .join(' ');
         const options: ExecOptions = {
             cwd: target.directory,
             maxBuffer: TEN_MEGA_BYTE
@@ -125,7 +132,7 @@ export class LocalInstaller extends EventEmitter {
     }
 
     private packOne(directory: string): Promise<void> {
-        return exec(`npm pack ${directory}`, { cwd: this.uniqueDir, maxBuffer: TEN_MEGA_BYTE })
+        return exec(`npm pack ${quotify(directory)}`, { cwd: this.uniqueDir, maxBuffer: TEN_MEGA_BYTE })
             .then(() => void this.emit('packed', directory));
     }
 

--- a/test/unit/LocalInstallerSpec.ts
+++ b/test/unit/LocalInstallerSpec.ts
@@ -50,9 +50,9 @@ describe('LocalInstaller install', () => {
 
         it('should pack correct packages', async () => {
             await sut.install();
-            expect(execStub).calledWith(`npm pack ${resolve('b')}`, { cwd: tmpDir, maxBuffer: TEN_MEGA_BYTE });
-            expect(execStub).calledWith(`npm pack ${resolve('c')}`, { cwd: tmpDir, maxBuffer: TEN_MEGA_BYTE });
-            expect(execStub).calledWith(`npm pack ${resolve('/e')}`, { cwd: tmpDir, maxBuffer: TEN_MEGA_BYTE });
+            expect(execStub).calledWith(`npm pack "${resolve('b')}"`, { cwd: tmpDir, maxBuffer: TEN_MEGA_BYTE });
+            expect(execStub).calledWith(`npm pack "${resolve('c')}"`, { cwd: tmpDir, maxBuffer: TEN_MEGA_BYTE });
+            expect(execStub).calledWith(`npm pack "${resolve('/e')}"`, { cwd: tmpDir, maxBuffer: TEN_MEGA_BYTE });
         });
 
         it('should install correct packages', async () => {
@@ -157,7 +157,7 @@ describe('LocalInstaller install', () => {
         });
     });
 
-    const tmp = (file: string) => resolve(tmpDir, file);
+    const tmp = (file: string) => `"${resolve(tmpDir, file)}"`;
 
     const stubPackageJson = (recipe: { [directory: string]: string }) => {
         Object.keys(recipe).forEach((directory, i) => {

